### PR TITLE
Add setup time for ctf, koth and dkoth

### PR DESCRIPTION
**Pull request by PrOF-kk**
Although I'm fairly confident, we should do some testing, I'm not sure TeamGates work as intended now

### CODE REVIEW

File: Source/gg2/Objects/Overlays/Gamemodes/CTFHUD.events/User Event 13.xml [hunk 1]
```diff
@@ -17,6 +17,7 @@
 global.timeLimitMins = read_ubyte(global.deserializeBuffer);
 timeLimit=global.timeLimitMins*30*60;
 timer=read_uint(global.deserializeBuffer);
+global.setupTimer=read_ushort(global.deserializeBuffer);
```
[the remark this answers is not in the document]
Comment (PrOF-kk) [reply Source/gg2/Objects/Overlays/Gamemodes/CTFHUD.events/User Event 13.xml]: Yes it crashes, is it correct to change the 5 to 7 in this case? I can't find the right documentation on type size

**Comment by PrOF-kk** [comment]
Question: should shouldSetup use the [areSetupGatesClosed](https://github.com/Gang-Garrison-2/Gang-Garrison-2/blob/master/Source/gg2/Scripts/Physics/areSetupGatesClosed.gml) script instead of only checking for `global.setupTimer `?

File: Source/gg2/Objects/Overlays/Gamemodes/CTFHUD.events/User Event 13.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Medo42) [reply Source/gg2/Objects/Overlays/Gamemodes/CTFHUD.events/User Event 13.xml]: It should be in the Faucet Networking documentation, but yes, a short is two bytes. I admit the names could have been chosen more clearly. Like read_uint16_t ;)

**Comment by Medo42** [comment]
> Question: should shouldSetup use the [areSetupGatesClosed](https://github.com/Gang-Garrison-2/Gang-Garrison-2/blob/master/Source/gg2/Scripts/Physics/areSetupGatesClosed.gml) script instead of only checking for `global.setupTimer `?

Well... it's not an easy question. Right now it looks to me like we have two different notions of setup, one which affects the TeamGates and one which affects ControlPointSetupGates. And the two overlap, meaning that for maps with ControlPointSetupGates we now *additionally* lock the team gates, which might break some maps.

Perhaps we should be asking two questions: "Is setup active right now?" (meaning there can only be one concept of setup time), and "which entities become solid during setup time for the current map".

**Comment by PrOF-kk** [comment]
Rebased to fix merge conflicts

**Comment by PrOF-kk** [comment]
After the dkoth crashes from 2.10.0-beta1 I took a look at the code again. While the PR manages to be a fairly simple "plug-and-play" solution, it's still pretty messy,  doesn't add 10 seconds of setup time on CP maps without setup gates, and it's not clear from the surrounding code in the event what the "doCheckSetupTime" script actually does.  
I'm closing this for now, hopefully someday I'll have enough time to write a better implementation

**Pull request closed without merging**